smrf/tests/test_distribute_permutations.py

```diff
@@ -0,0 +1,92 @@
+from unittest.mock import patch, MagicMock
+from inicheck.tools import cast_all_variables
+import numpy as np
+from parameterized import parameterized
+
+from smrf.tests.mock_helpers import SmrfMocks
+from smrf.tests.smrf_test_case import SMRFTestCase
+import smrf
+
+
+class ConfigPermutations:
+    def __init__(self, config_changes, distribute_section, variable, expected_array):
+        self.config_changes = config_changes
+        self.distribute_section = distribute_section
+        self.variable = variable
+        self.expected_array = expected_array
+
+
+class TestDistributePermutations(SMRFTestCase):
```

Comment:
The idea here is that we can track that an expected change is happening to a variable distribution based on the changes in the config.

Reply:
I am also torn whether the level of complexity of the Mock is balanced with the benefit of the test.
Could you add a unit level test on the `precipitation` class and forgo the integration level test for the parameter? Or as Scott suggested, the integration level test mainly checks that the outputs compared to the Gold just be larger or smaller depending on the scale factor.

Reply:
I'm torn here as well. My initial thought was that we may want to add more complicated scalar methods down the line, like maybe an approach that scales the precip by elevation if that is the systematic error we know exists in the input data. With a more complicated scaling approach, I'm inclined to stay away from the gold files and lean into a 4x4 array that we can specify in the test.

My current approach does feel like a mix and match of integration and unit testing, though. Maybe a pure unit test like @jomey suggested with all the data mocked out would be a better tact. This could be a better testing paradigm for testing more permutations of the config.

Thoughts?

Reply:
I can see this being added upon as well in the future so we should think about the best implementation for this.

I'm a big fan of integration testing, especially when we are talking about modifying outputs. Then this needs to be tested again the gold files. While I see the benefit of having a smaller subet to work with so you can actually see the results, a small domain does now capture a large enough sample size. We have already seen it with the RME test case where it cannot pick up issues with the code because it's too small, which is why we moved to the Lakes. If it takes even 1 more minute to run the tests then that is an okay tradeoff.

With that said, I agree that a few more unit tests would be useful. In my opinion, the integration test should test against the gold file in the Lakes to ensure that we back out the correct precipitation field. This will ensure that nothing else is being introduced from the scaling factors.

Reply:
@scotthavens I refactored the test to be an integration test, checking the ratio of the gold file to the test file. It is currently setup with the RME test case. I'm happy to swap it over to use Lakes, but the Lakes tests fail for me locally (maybe a Mac thing?) so I'm sticking with RME until the test methodology is approved.

Reply:
@jomey implemented an environment variable `NOT_ON_GOLD_HOST` which will relax the test requirements. It should pass on the CI actions when running with Ubuntu.

Reply:
I had to brew install eccodes. That created a sneaky runtime error that didn't show up in the logs at all. About to push again with the test changed to use the Lakes data